File: Data Structures & Algorithms/two-integer-sum/submission-7.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int i =0;i<nums.length;i++){
            map.put(nums[i],i);
        }
        int[] res = new int[2];
        for (int j =0;j<nums.length;j++){
            int diff = target-nums[j];
            if ((map.containsKey(diff))&&map.get(diff)!=j){
                if (map.get(diff)>j){
                    res[0]=j;
                    res[1]=map.get(diff);
                }
                else if (map.get(diff)<j){
                    res[1]=j;
                    res[0]=map.get(diff);
                }
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-7.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        res = nums[0]

        l, r = 0, len(nums) - 1

        while l <= r:
            if nums[l] < nums[r]:
                return min(res, nums[l])
            mid = (r + l) // 2
            res = min(res, nums[mid])
            if nums[l] <= nums[mid]:
                l += 1
            else:
                r -= 1
        return res